src/video/VDP.cc

```diff
@@ -1931,7 +1949,8 @@ void VDP::serialize(Archive& ar, unsigned serVersion)
 	             "blinkState",           blinkState,
 	             "dataLatch",            dataLatch,
 	             "registerDataStored",   registerDataStored,
-	             "paletteDataStored",    paletteDataStored);
+	             "paletteDataStored",    paletteDataStored,
+	             "writeAccess",          writeAccess);
```

Comment:
Unfortunately this is insufficient:
As-is this change will break loading existing savestate files: it will fail with an error like `"missing tag <writeAccess>"`.
To properly implement this you must:
* Around line 1882: add a comment like `// version 10: added writeAccess`.
* In VDP.hh around line 1281: increase the version from 9 to 10.
* Around line 1972: add a block like:
```
if (ar.versionAtLeast(setVersion, 10)) {
    ar.serialize("writeAccess", writeAccess);
} else {
    writeAccess = !cpuVramReqIsRead; // best guess
}
```

The comment describes how the savestate format has evolved in the different versions. So that we don't have to go digging in git-history for this information. We then increase the version number from 9 to 10. These version numbers are local to the VDP object itself (so the full savestate has many sub-version-numbers for each of the different components). And then we have a code block that deals with loading version 10 (and higher) and versions 9 (and lower). For the lower versions we can't 100% correctly initialize `writeAccess`, so we make a best-effort attempt. (Note that at this point `cpuVramReqIsRead` must already be restored from the savestate, the order of these things is important).

For backwards compatibility, this "version 10" of the VDP savestate format is something we need to keep supporting forever. That's also a reason why I would have preferred to not introduce extra member variables only for a debuggable.

Reply:
I've tested in on my machine. Working fine.